control/HTTP.php

```diff
@@ -392,10 +392,22 @@ public static function add_cache_headers($body = null) {
 		if (!$config->get('cache_ajax_requests') && function_exists('apache_request_headers')) {
 			$requestHeaders = array_change_key_case(apache_request_headers(), CASE_LOWER);
 			if (array_key_exists('x-requested-with', $requestHeaders) && strtolower($requestHeaders['x-requested-with']) == 'xmlhttprequest') {
-				$cacheControl->disableCache(true);
+				$cacheControl->disableCache();
```

Comment:
This seems appropriate, `cache_ajax_requests` is a (less common) default you might want to overrule. We should deprecate this setting anyway for 4.x, I've added a subtask for that.